rust/src/datafusion/logical_expr.rs

```diff
@@ -116,6 +127,64 @@ pub fn resolve_expr(expr: &Expr, schema: &Schema) -> Result<Expr> {
     }
 }
 
+/// Coerce logical expression for filters to bollean.
+///
+/// Parameters
+///
+/// - *expr*: a datafusion logical expression
+pub fn coerce_filter_type_to_boolean(expr: &Expr) -> Result<Expr> {
+    match expr {
+        // I don't think that datafusion has functions for operating on lists, so we need to make UDFs
```

Comment:
Instead of the UDF, could we wrap in `IS NOT NULL`? The docs say if there is no match then the result will be null. https://docs.rs/arrow-string/40.0.0/arrow_string/regexp/fn.regexp_match.html

But there's definitely a more optimal route we should look into in the future, tracked in this issue: https://github.com/apache/arrow-rs/issues/3568